### PR TITLE
Remove code duplication of loadModule in binaryen

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: src/binaryen.h [hunk 1]
```diff
@@ -41,6 +41,10 @@ class BinaryenEngine : public WasmEngine {
 
 private:
   void verifyContract(wasm::Module & module);
+
+  /// Parses and loads a Wasm module.
+  /// Don't ask, Module has no copy constructor, hence the reference.
+  void loadModule(std::vector<uint8_t> const& code, wasm::Module & module);
```
Comment (jakelang) [inline src/binaryen.h]: Perhaps we could move this to a virtual method so that any abstract VM should have to implement it?

Comment (axic) [reply src/binaryen.h]: It would be super abstract 😉

File: src/binaryen.cpp [hunk 2]
```diff
@@ -517,10 +509,8 @@ ExecutionResult BinaryenEngine::execute(
   return result;
 }
 
-void BinaryenEngine::verifyContract(vector<uint8_t> const& code)
+void BinaryenEngine::loadModule(vector<uint8_t> const& code, wasm::Module & module)
```
Comment (chfast) [inline src/binaryen.cpp]: Can't you return it?

Comment (axic) [reply src/binaryen.cpp]: See the comment in the header:
> /// Don't ask, Module has no copy constructor, hence the reference.

**Pull request merged**